### DISCUSSION TITLE
New architecture support

### DISCUSSION THREAD

**Discussion by PopJoestar** [lodev09/react-native-true-sheet#20] [General]
Hello, does this work with new architecture?

**Comment by lodev09** [comment]
Not yet unfortunately. There are some blockers on interop layer when using swift. Or I'm missing something.
Either way, just gonna hold on for a bit until new arc becomes stable

**Reply by Martin2037** [reply]
Really like this project💗
New arch works fine with swift, check out [turbo swift](https://github.com/react-native-community/RNNewArchitectureLibraries/tree/feat/turbomodule-swift)

Need New Arch Support～～～

**Reply by lodev09** [reply]
Yeah it can run on swift with new arch and bridgeless mode enabled but the touch handler is not working 😢 

Also, it needed Fiber support and Swift isn't supported yet.

<img width="875" alt="Screenshot 2024-05-11 at 19 21 20" src="https://github.com/lodev09/react-native-true-sheet/assets/6686328/513c8e6f-6692-4a02-8731-2c5544210f7c">

The only option at this point is to convert the whole IOS lib to Objective-C for it to work with the interop layer and ultimately support FIber. So I'm holding on for a bit, maybe they will release support Swift soon

**Reply by cloudorbush** [reply]
> Yeah it can run on swift with new arch and bridgeless mode enabled but the touch handler is not working 😢 
> 
> 
> 
> Also, it needed Fiber support and Swift isn't supported yet.
> 
> 
> 
> <img width="875" alt="Screenshot 2024-05-11 at 19 21 20" src="https://github.com/lodev09/react-native-true-sheet/assets/6686328/513c8e6f-6692-4a02-8731-2c5544210f7c">
> 
> 
> 
> The only option at this point is to convert the whole IOS lib to Objective-C for it to work with the interop layer and ultimately support FIber. So I'm holding on for a bit, maybe they will release support Swift soon

Hmm that's unfortunate. This was my favorite modal library for react native 😭

**Reply by lodev09** [reply]
Yeah. Can't do anything about it since it's a missing feature from RN. I wanted to support new arch badly. If next version does not support it still, I'll have to convert ios to objective-c :D

**Comment by lodev09** [comment]
Created [issue](https://github.com/facebook/react-native/issues/45232) at RN. hopefully they can give more insight. 🙏

**Reply by lodev09** [reply]
Good news.. looks like we will be supporting New Arch really soon! I'll just wait for expo to update RN with the guy's PR 🚀

**Comment by gorbypark** [comment]
I'm playing around with the Expo SDK52 beta with new arch enabled by default.  Came across this because buttons/pressables are not working 😅 

Do we know if the PR to the interop layer on the RN side of things landed?  Expo 52 is using react native 0.76.0.

**Reply by lodev09** [reply]
Yep. There's a fix for that regarding the touchable button. My only issue at this time is the sizing.. Haven't had a chance to try it yet. Hopefully we don't end up rewriting the library to obj-c :(

**Reply by gorbypark** [reply]
Without having actually done much investigating besides reading the linked react native issue, I wonder if a port to something like nitro modules might be easier than a rewrite to obj-c?  I haven't actually used nitro modules yet, but it seems like a decent alternative to an expo-module.  An expo module would be a good choice too, however that would limit usage to expo-based project (of course you can always install expo into a non-expo project, but sometimes that's undesirable).

**Reply by akuul** [reply]
I fail to understand if the interop layer was supposed to fix this issue ? React Native pressables are still not working for me.

**Reply by lodev09** [reply]
There's a native method that the interop layer (or RN) does not support and causing sizing issue. The pressables has a fix but haven't updated TrueSheet yet due to that size issue. I was kinda hoping it fixes itself after some time but I guess it's not that easy :(

As mentioned in that thread, I'll try to fix the size on the JS side but I'm not really confident about it to be honest.

**Reply by akuul** [reply]
I have moved to RNGH RectButton for the time being. Honestly, might just stay with it since it works perfectly fine with my use case. Leaving this here in case some one wants to use TrueSheet with New Architecture and want to explore RNGH alternatives.

**Reply by lodev09** [reply]
@akuul just curious, are you not having sizing issue on your end?

**Reply by akuul** [reply]
If by sizing you mean the transparent bottom then yes, I do have it. I just applied padding as a very quick and stupid workaround for now. Just really didn't want to opt-out from New Arch and TrueSheet. While far from ideal - works for now.

**Reply by dhruvpvx** [reply]
Is there any workaround for it?

**Reply by ReactorCSA** [reply]
> There's a native method that the interop layer (or RN) does not support and causing sizing issue. The pressables has a fix but haven't updated TrueSheet yet due to that size issue. I was kinda hoping it fixes itself after some time but I guess it's not that easy :(
> 
> As mentioned in that thread, I'll try to fix the size on the JS side but I'm not really confident about it to be honest.

Hi @lodev09 !

Maybe you could deploy the pressables fix? Sizing issue seems to be not as critical as non working pressables, since sizing may not be used at all if sheet is fullScreen, but pressables are really critical issue :(

Or if it's possible maybe you could provide a patch file so we can locally apply it?

Thanks in advance!

**Reply by lodev09** [reply]
yep, working on it this weekend

**Reply by ReactorCSA** [reply]
Thank you!!!!!!!!!!!!

**Reply by ReactorCSA** [reply]
Hi @lodev09!

How is the progress fixing Touchables on Android? Sorry for pushing but I am really waiting the fix so hard

**Reply by lodev09** [reply]
I can only work on it on my free time. I'm kinda busy with other things :/
Sorry.

**Reply by ReactorCSA** [reply]
Hi @lodev09 !

Do you have any progress regarding adapting your library to new arch?

**Reply by dinosmajovic** [reply]
So excited to use it with new arch

**Reply by lodev09** [reply]
Hey, please check PR #106. I've made some changes to this lib to work on New Arch thru the interop layer. It's not fully Fabric yet (see discussion above) but hopefully this works in the meantime!

Thanks!

**Comment by sajorahasan** [comment]
With new architecture enabled in expo 52 stable, I keep getting app crash on android

Here is crash log:
```

msg: java.lang.UnsupportedOperationException: EventEmitter#receiveTouches is not supported by the Fabric Interop Layer
--
  | stacktrace: java.lang.UnsupportedOperationException: EventEmitter#receiveTouches is not supported by the Fabric Interop Layer
  | at com.facebook.react.internal.interop.InteropEventEmitter.receiveTouches(InteropEventEmitter.kt:47)
  | at com.facebook.react.uimanager.events.TouchesHelper.sendTouchesLegacy(TouchesHelper.java:107)
  | at com.facebook.react.uimanager.events.ReactEventEmitter.receiveTouches(ReactEventEmitter.java:92)
  | at com.facebook.react.uimanager.events.TouchEvent.dispatchModern(TouchEvent.kt:124)
  | at com.facebook.react.uimanager.events.FabricEventDispatcher.dispatchEvent(FabricEventDispatcher.java:76)
  | at com.facebook.react.uimanager.JSTouchDispatcher.handleTouchEvent(JSTouchDispatcher.java:97)
  | at com.facebook.react.uimanager.JSTouchDispatcher.handleTouchEvent(JSTouchDispatcher.java:68)
  | at com.lodev09.truesheet.core.RootSheetView.onInterceptTouchEvent(RootSheetView.kt:102)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2659)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3125)
  | at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2749)
  | at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:496)
  | at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1953)
  | at android.app.Dialog.dispatchTouchEvent(Dialog.java:933)
  | at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:70)
  | at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:454)
  | at android.view.View.dispatchPointerEvent(View.java:16548)
  | at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:7586)
  | at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:7348)
  | at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6745)
  | at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6802)
  | at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6768)
  | at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:6934)
  | at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6776)
  | at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:6991)
  | at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6749)
  | at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6802)
  | at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6768)
  | at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6776)
  | at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6749)
  | at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:9854)
  | at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:9805)
  | at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:9774)
  | at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:9996)
  | at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:295)
  | at android.os.MessageQueue.nativePollOnce(Native Method)
  | at android.os.MessageQueue.next(MessageQueue.java:349)
  | at android.os.Looper.loopOnce(Looper.java:189)
  | at android.os.Looper.loop(Looper.java:317)
  | at android.app.ActivityThread.main(ActivityThread.java:8592)
  | at java.lang.reflect.Method.invoke(Native Method)
  | at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
  | at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)


```

**Comment by ferdinanddoremus** [comment]
Thanks @lodev09 , any plan on officially releasing it since it is merged on main now?

**Reply by lodev09** [reply]
I'm currently testing it on one of my production app. I'll release once it's stable :)

**Reply by codydorsettlynn** [reply]
Any chance of a canary tagged release in the meantime?

**Reply by ReactorCSA** [reply]
Currently I'm testing it directly applying dependency as follows:
"@lodev09/react-native-true-sheet": "lodev09/react-native-true-sheet#main",
It works completely fine! @lodev09 thank you for your hard work!

**Comment by lodev09** [comment]
This is now released! Thank you all for the feedback 🔥
Hopefully Fiber gets supported in the future 🙏
